libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/verification/RustSessionVerificationService.kt

```diff
@@ -188,12 +195,21 @@ class RustSessionVerificationService(
         _verificationFlowState.value = VerificationFlowState.Initial
     }
 
+    override suspend fun saveVerifiedState(verified: Boolean) = tryOrFail {
+        val existingSession = sessionStore.getSession(client.userId())
+            ?: error("Failed to save verification state. No session with id ${client.userId()}")
+        sessionStore.updateData(existingSession.copy(needsVerification = !verified))
+        // Wait until the new state is saved
```

Comment:
Why we need to wait here?

Reply:
Because otherwise if you go back immediately to any of the screens that use `needsVerificationFlow`, the state is still `true`. I found it odd too since I expected the new value to be emitted by the time `sessionStore.updateData` is finished, but that didn't seem to be the case.